### PR TITLE
Fixes deadlock in retry

### PR DESCRIPTION
**Pull request by garrettmoon**
Also adds a test to hopefully prevent this in the future.

Also made it so the timeout value of the session configuration is
respected instead of having an internal unexposed timeout.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f06da00e8de15636a30a2712a0cebaf0/raw/ba94f06b6dd0fd140bb52827937cc2ea3d4df7c3/log.txt)

File: Source/Classes/PINRemoteImageManager.m [hunk 1]
```diff
@@ -878,7 +879,7 @@ - (NSURLSessionDataTask *)dataTaskWithURL:(NSURL *)url
 {
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url
                                                            cachePolicy:NSURLRequestReloadIgnoringLocalCacheData
-                                                       timeoutInterval:self.timeout];
```
Comment (nguyenhuy) [inline Source/Classes/PINRemoteImageManager.m]: Let's remove `timeout` property.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/004038336a9017ea3b00dfd2b3c73c47/raw/c241e908e2924bfc46a3341e1a289e3ac846a471/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/edd146b5ce63e71060db8c291759ceab/raw/1f197d074dc30d7497ca2d1b5ec2fae484bbf604/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/a0dbc21bf20d5c9575f3d6f0f7d36f00/raw/ad8f1adc6f55c0f3317e10c7d8986a8b4a9a94a2/log.txt)

**Pull request merged**